src/main/java/org/folio/util/TokenCookieParser.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.util;
+
+import org.apache.commons.lang3.StringUtils;
+import org.folio.rest.impl.LoginAPI;
+
+public class TokenCookieParser {
+  public String getAccessToken() {
+    return accessToken;
+  }
+  private String accessToken;
+
+  public String getRefreshToken() {
+    return refreshToken;
+  }
+  private String refreshToken;
+
+  public TokenCookieParser(String cookieHeader) {
+    accessToken = getCookieValue(cookieHeader, LoginAPI.ACCESS_TOKEN);
+    refreshToken = getCookieValue(cookieHeader, LoginAPI.REFRESH_TOKEN);
+  }
+
+  private String getCookieValue(String cookieHeader, String name) {
+    String headerStripped = StringUtils.strip(cookieHeader, "\t");
+    String[] cookies = headerStripped.split(";");
+
+    if (cookieNameOccursMoreThanOnce(cookies, name)) {
+      throw new IllegalArgumentException("Cookie name occurs more than once: " + name);
+    }
+
+    for (String c : cookies) {
+      String[] cookeNameValue = getCookieNameValue(c);
+      String cookieName = cookeNameValue[0].trim();
+      String cookieValue = cookeNameValue[1].trim();
+      if (cookieName.equals(name)) {
+        return cookieValue;
+      }
+    }
+
+    throw new IllegalArgumentException("No cookie found for name: " + name);
+  }
+
+  private String[] getCookieNameValue(String c) {
+    return c.split("=", 2);
+  }
+
+  private boolean cookieNameOccursMoreThanOnce(String[] cookies, String cookieName) {
+    int count = 0;
+
+    for (String c : cookies) {
+      String[] cookeNameValue = getCookieNameValue(c);
+      if (cookieName.equals(cookeNameValue[0])) {
+        count++;
+      }
+    }
+
+    return count > 1;
+  }
+}
```

Comment:
This is security sensitive. Please add unit tests for TokenCookieParser.

Reply:
I added some unit tests in a new class called `CookieTests`. I think maybe I've got the OWS sorted out. See what you think.

Reply:
I also removed the usage of the `StringUtils`. It seems like `trim` takes care of what we want no?

Reply:
Is the trimming of `ServerCookieDecoder.STRICT.decodeAll` incomplete? Is there still a need that we do some additional `trim`? If yes please provide a unit test for this case.

Reply:
Otherwise remove the `trim`.

Reply:
See the new test `testParseCookieWithWhitespaceTabAndLF`. Without the `trim` on `cookieHeader` at line 29 this test fails. In other words, `ServerCookieDecoder.STRICT.decodeAll` doesn't trim the OWS. I guess they're leaving that up to us to decide if the whitespace has meaning or not.

Reply:
Thanks!